django/db/migrations/state.py

```diff
@@ -83,6 +83,9 @@ def remove_model(self, app_label, model_name):
         del self.models[app_label, model_name]
         if 'apps' in self.__dict__:  # hasattr would cache the property
             self.apps.unregister_model(app_label, model_name)
+        # Need to do it once all models are unregistered to avoid corrupting
+        # existing models' _meta
+        self.apps.clear_cache()
```

Comment:
This needs to be on the same indentation level as `unregister_model()`, otherwise you're enforcing a rendering of `self.apps` ;)